runtime/src/snapshot_utils.rs

```diff
@@ -1131,9 +1790,82 @@ pub fn bank_to_snapshot_archive<P: AsRef<Path>, Q: AsRef<Path>>(
     Ok(package.tar_output_file)
 }
 
+/// Convenience function to create an incremental snapshot archive out of any Bank, regardless of
+/// state.  The Bank will be frozen during the process.
+///
+/// Requires:
+///     - `bank` is complete
+///     - `bank`'s slot is greater than `full_snapshot_slot`
+pub fn bank_to_incremental_snapshot_archive<P: AsRef<Path>, Q: AsRef<Path>>(
+    snapshot_path: P,
+    bank: &Bank,
+    full_snapshot_slot: Slot,
+    snapshot_version: Option<SnapshotVersion>,
+    snapshot_package_output_path: Q,
+    archive_format: ArchiveFormat,
+    thread_pool: Option<&ThreadPool>,
+    maximum_snapshots_to_retain: usize,
+) -> Result<PathBuf> {
+    let snapshot_version = snapshot_version.unwrap_or_default();
+
+    assert!(bank.is_complete());
```

Comment:
Maybe we can put this common code into a separate function to share with bank_to_snapshot_archive?

Reply:
Yeah, I thought about doing that too. Unfortunately the only real common code are the first few lines of `bank.xxx()`, then afterwards they are different depending on full snapshot vs incremental snapshot. I decided to leave them separate for that reason, as the duplication wasn't much. Also, I am anticipating needing to change the impl in the incremental snapshot side w.r.t. clean_accounts(), so that's another reason I've left them separate. Is that OK for now?